packages/diagnostics_channel/src/index.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const {
+  Channel,
+  channel
+} = require('diagnostics_channel') // eslint-disable-line n/no-restricted-require
+
+const [major, minor] = process.versions.node.split('.')
+
+// Our own DC with a limited subset of functionality stable across Node versions.
+// TODO: Move the rest of the polyfill here.
+// TODO: Switch to using global subscribe/unsubscribe/hasSubscribers.
+const dc = { channel }
+
+// Prevent going to 0 subscribers to avoid bug in Node.
+// See https://github.com/nodejs/node/pull/47520
+if (major === '19' && minor === '9') {
+  const channel = dc.channel
+
+  dc.channel = function () {
+    const maybeInactive = channel.apply(this, arguments)
+
+    if (maybeInactive.subscribe === Channel.prototype.subscribe) {
+      const subscribe = maybeInactive.subscribe
+
+      maybeInactive.subscribe = function () {
+        delete maybeInactive.subscribe
+
+        subscribe.apply(this, arguments)
+
+        this.subscribe(() => {}) // Keep it active forever.
+      }
+    }
```

Comment:
If it's already active, we still need to be sure there's an extra subscriber forever.

```suggestion
    } else {
      maybeInactive.subscribe(() => {}) // Keep it active forever.
    }
```

Reply:
How could it be active? It would have to first be inactive before it's ever active.

Reply:
All tests passing also prove that this is the case, and preventing usage of the built-in with eslint ensures that nothing subscribes with the built-in directly.

Reply:
If something else subscribes to that same named channel before we get to this point it will already be active. Unlikely given we're currently using our own custom channel names, but possible. And it will become more likely as we start pushing instrumentations upstream.

Reply:
> If you're going to use the WeakMap, why even patch the subscribe function?

Because otherwise a subscriber would be added on `channel()` and not on `subsribe()`, so publishers would cause a subscription as well.

> Also, the return value of the patched subscribe is wrong, it should return the result of the subscribe call.

True, I didn't return anything because the method doesn't either, but it's best practice to return anyway in case that changes.

Reply:
Ah, right, it's only unsubscribe that returns a value. Misremembering my own code. 😆